### DISCUSSION TITLE
Improve file upload in FileElement

### DISCUSSION THREAD

**Discussion by adamberecz** [vueform/vueform#49] [Ideas]
- Option to send files only on form submit instead of temp upload

**Comment by rriemann** [comment]
I have the same question. I read about https://axios-http.com/docs/multipart as vueform/vue seem to rely on axios. In axios, it is possible. So I assume that it just requires an appropriate method for `upload-temp-endpoint`.

Does anyone has a working example?

**Reply by adamberecz** [reply]
To skip temp upload and send all the files with the final form request, you can do this:

```js
file: {
  type: 'file',
  uploadTempEndpoint: f => f,
}
```

The `uploadTempEndpoint` will return the `File` object as it is instead of uploading thus keeping it as the element value, which will then get submitted with the form later on.

I needed to add a small patch to make this work, so make sure to upgrade to `1.7.3` before testing.

**Reply by rriemann** [reply]
Very much appreciated! In my local environment, tests have been successful so far!

**Reply by rriemann** [reply]
For consistancy, I would like to propose a feature request.

I want to configure the endpoint globally in my `vueform.config.js`:

~~~js
export default defineConfig({
  endpoints: {
    // https://github.com/vueform/vueform/discussions/49
    uploadTempFile: f => f,
  },
})
~~~

However, doing so, gives me currently an error:

~~~
Uncaught (in promise) Error: TypeError: oe.value.uploadTempFile.method is undefined
~~~

The multifileelement property accepts the handler `f => f`, but the vueform config does not.

**Comment by maxwellsmart84** [comment]
I had this working as it is a feature of the component, however I agree the method in getting it to work is a little strange as it's based on the return value of your async upload function:

<img width="936" alt="image" src="https://github.com/vueform/vueform/assets/14876902/31ac581a-9bb8-446f-b4cb-fdd9264e98d1">

**Comment by adamberecz** [comment]
Since creating this Discussion imo file uploading has improved enough to be able to cover most of the use cases. I've also recently created a separate docs on it: https://vueform.com/docs/file-uploads

I'm closing this now, feel free to reopen if you have a use-case that's not covered by the current feature set.
